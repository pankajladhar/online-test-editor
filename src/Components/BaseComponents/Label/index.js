import React from "react";
import PropTypes from "prop-types";

const Label = props => {
  const { htmlFor, className, text } = props;
  return (
    <label htmlFor={htmlFor} className={`Label ${className}`}>
      {text}
    </label>
  );
};

Label.defaultProps = {
  className: ""
};

Label.propTypes = {
  htmlFor: PropTypes.string.isRequired,
  className: PropTypes.string,
  text: PropTypes.string.isRequired
};

export default Label;
