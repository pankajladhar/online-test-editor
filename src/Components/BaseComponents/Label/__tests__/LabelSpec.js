import React from 'react';
import renderer from 'react-test-renderer';
import Label from './../index';

describe('Label', ()=>{
    let labelProps = {};

    beforeEach(()=>{
        labelProps = {
            htmlFor: "htmlFor",
            text: "some text"
        };
    })

    it('renders correctly with mandatory params', ()=>{
        let tree = renderer.create(
            <Label {...labelProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with optional params', ()=>{
        labelProps.className = "someClass"
        let tree = renderer.create(
            <Label {...labelProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});