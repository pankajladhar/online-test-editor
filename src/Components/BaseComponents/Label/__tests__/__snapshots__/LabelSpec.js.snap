// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly with mandatory params 1`] = `
<label
  className="Label "
  htmlFor="htmlFor"
>
  some text
</label>
`;

exports[`Label renders correctly with optional params 1`] = `
<label
  className="Label someClass"
  htmlFor="htmlFor"
>
  some text
</label>
`;
